client/ipa-getkeytab.c

```diff
@@ -137,26 +158,48 @@ int filter_keys(krb5_context krbctx, struct keys_container *keys,
     return n;
 }
 
-static int ipa_ldap_init(LDAP ** ld, const char * scheme, const char * servername, const int  port)
+static int ipa_server_to_uri(const char *servername, const char *mech,
+                             char **ldap_uri)
 {
-	char* url = NULL;
-	int  url_len = snprintf(url,0,"%s://%s:%d",scheme,servername,port) +1;
+    char *url = NULL;
+    int url_len = 0;
+    int port = 0;
+    char *scheme = NULL;
+
+    if (!mech) {
```

Comment:
That is probably some leftover from previous incarnations of the patch.  If I manage to make StartTLS work this may actually be irrelevant since we may use ldap://$FQDN and have secure connection.